src/main/java/htsjdk/samtools/read/Read.java

```diff
@@ -0,0 +1,1438 @@
+     *
+     * @return the read name or {@code null} if the read has no name.
+     */
+    String getName();
+
+    /**
+     * Sets the name of the read.
+     *
+     * <p>Equivalent to <b>QNAME</b> in the <a href="http://samtools.github.io/hts-specs/SAMv1.pdf">SAM specifications</a>.
+     *
+     * @param name new name for the read; {@code null} if the read has no name.
+     */
+    void setName(final String name);
+
+    /**
+     * Gets the bitwise flag.
+     *
+     * <p>Equivalent to <b>FLAG</b> in the <a href="http://samtools.github.io/hts-specs/SAMv1.pdf">SAM specifications</a>.
+     *
+     * <p>Note: it is preferable to use the methods for testing if specific flags are set (using {@link #isSet(SAMFlag)} or {@link #isUnset(SAMFlag)}),
+     * get all the set flags with {@link #getSetFlags()} or testing specific flags with the shortcuts.
+     *
+     * @return the bitwise flag as an integer.
+     */
+    int getFlag();
+
+    /**
+     * Gets the {@link SAMFlag} that are set.
+     *
+     * <p>Equivalent to <b>FLAG</b> in the <a href="http://samtools.github.io/hts-specs/SAMv1.pdf">SAM specifications</a>.
+     *
+     * <p>Default implementation calls {@code SAMFlag.getFlags(getFlag())}.
+     *
+     * @return a set of the flags that are set for the read.
+     */
+    default Set<SAMFlag> getSetFlags() {
+        return SAMFlag.getFlags(getFlag());
+    }
+
+    /**
+     * Checks if the flag is set.
+     *
+     * <p>Default implementation calls {@code flag.isSet(getFlag())}.
+     *
+     * @param flag the flag to test.
+     * @return {@code true} if the flag is set; {@code false} otherwise.
+     *
+     * @throws IllegalArgumentException if the flag is {@code null}.
+     */
+    default boolean isSet(final SAMFlag flag) {
+        return flag.isSet(getFlag());
+    }
+
+    /**
+     * Checks if the flag is unset.
+     *
+     * <p>Default implementation calls {@code flag.isUnset(getFlag())}.
+     *
+     * @param flag the flag to test.
+     * @return {@code true} if the flag is unset; {@code false} otherwise.
+     *
+     * @throws IllegalArgumentException if the flag is {@code null}.
+     */
+    default boolean isUnset(final SAMFlag flag) {
+        return flag.isUnset(getFlag());
+    }
+
+    /**
+     * Sets the bitwise flag.
+     *
+     * <p>Equivalent to <b>FLAG</b> in the <a href="http://samtools.github.io/hts-specs/SAMv1.pdf">SAM specifications</a>.
+     *
+     * <p>Note: it is preferable to use the methods for set/unset specific flags are set (using {@link #setFlag(SAMFlag)} or {@link #unsetFlag(SAMFlag)}),
+     * clear the flags using {@link #clearFlag()}, set/unset the specific flags (using {@link #setFlags(Set) and {@link #unsetFlags(Set)}},
+     * or use the shortcuts.
+     *
+     * @param flag bitwise flag to assign to the read.
+     */
+    void setFlag(final int flag);
```

Comment:
there should also be an unsetFlag, no?

Reply:
oh, I see this is for setting all the bits...

I think we should have a nomenclature that distinguishes between a single bit and the entires set of bits. I would advocate that "flag" (singular) refers to a bit, and "flags" (plural) refers to the entire integer that is all the bits.

Reply:
and thus, would call this set Flags.